Marlin/stepper.cpp

```diff
@@ -368,6 +382,11 @@ void Stepper::isr() {
         if (current_block->steps[Z_AXIS] > 0) {
           enable_z();
           OCR1A = 2000; // Run at slow speed - 1 KHz
+          #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
+            SBI(TIMSK0, OCIE0A);
+          #endif
+          SBI(TIMSK0, OCIE0B);
+          ENABLE_STEPPER_DRIVER_INTERRUPT();
```

Comment:
In fact, I guess it could. But I see no problem with that, then it should execute one advance ISR, jump back and finish the return of the stepper ISR.
If that's a problem, a cli() before enabling the ISR again should do the trick. Then the ISR can not start as long as the stepper ISR hasn't returned.